test/jtreg-ext/requires/VMProps.java

```diff
@@ -462,7 +462,29 @@ protected String vmCDSForCustomLoaders() {
      * @return true if this VM can write Java heap objects into the CDS archive
      */
     protected String vmCDSCanWriteArchivedJavaHeap() {
-        return "" + ("true".equals(vmCDS()) && WB.canWriteJavaHeapArchive());
+        return "" + ("true".equals(vmCDS()) && WB.canWriteJavaHeapArchive()
+                     && isCDSRuntimeOptionsCompatible());
+    }
+
+    /**
+     * @return true if the VM options specified via the "test.cds.runtime.options"
+     * property is compatible with writing Java heap objects into the CDS archive
+     */
+    protected boolean isCDSRuntimeOptionsCompatible() {
+        String jtropts = System.getProperty("test.cds.runtime.options");
+        if (jtropts == null)
+            return true;
+        String CCP_DISABLED = "-XX:-UseCompressedClassPointers";
+        for (String opt : jtropts.split(",")) {
+            if (opt.equals(CCP_DISABLED))
+                return false;
+            if (opt.startsWith(GC_PREFIX) && opt.endsWith(GC_SUFFIX)) {
+                String gc = opt.substring(GC_PREFIX.length(), opt.length() - GC_SUFFIX.length());
+                if (!gc.equals("G1"))
+                    return false;
```

Comment:
Can't you just check for `-XX:+UseG1GC` ?

Reply:
I've pushed another commit based on your suggestion.